### DISCUSSION TITLE
Livewire Component Classifier

### DISCUSSION THREAD

**Discussion by ryanshipswell** [stefanzweifel/laravel-stats#199] [Ideas]
I made a Livewire Component Classifier that I am planning to submit as a PR. Before I finalize it I want some feedback.

For those who don't know Livewire has two different types of components. A full page component where a route points directly to the component class and a inline component which is referenced from within the blade file for an existing view or component.

In my testing it seems a full page component is picked up as a Controller by Laravel Stats by defualt, because it is pointed to directly by a route. But in-line components are ignored. I wrote a classifier to just looks for a class that extends `Livewire\Component` which will pickup in line components, but full page components are already classified as a controller.

So I have 3 options:

1) Submit the PR as-is. Full page Livewire components will stay classified as a Controller (as they are today without a custom classifier) and my new classifier will classify any Inline Components only as a "Livewire Component".

2) Prioritize my classifier over the Controller classifier so that both full page and inline components are classified as one type "Livewire Component" and the controller classifier won't consider a full page Livewire component as a controller.

3) Make two different Livewire classifiers, one for full page and one for inline components (I would have to priorotize the full page component classifier over the controller classifier so that it will pickup those components before the controller classifier does.

Any of these options seems 'valid' to me, but I am leaning toward 1 since that maintains the status quo for defining a controller. But I can see an argument that a full page component shouldn't be consider a "controller". So I am curious to hear what anyone else thinks before committing to one way over the others.

**Comment by stefanzweifel** [comment]
I personally would prefer option 1 too. 

In my projects, I haven't used full page Livewire components yet. If I would use them, I think it would confuse me if those would be classified as Controllers, eventhough they act as Controllers.

**Reply by ryanshipswell** [reply]
I'm not sure I understand. You say you prefer to leave them classified as controllers but you find that confusing? I kinda agree about the confusion and the more I think about it I'm thinking 2 is probably a better option. 

I don't really see any value in distinguishing between full page and inline for the stats. I'm thinking jet let everything be classified as a livewire component. Which means I either need to make the livewire classifier take priority over the controller classifier or make the controller classifier ignore livewire components. Giving priority to the livewire classifier is probably easier. I assume it's down to the order of the classifiers in the array.

**Reply by stefanzweifel** [reply]
@x7ryan It seem's I haven't read your original question right. Sorry for that! I thought option 1 was "Livewire component are always classified as Livewire components". 

After re-reading everything, option 2 might be better. Then a Livewire component will always be classified as a Livewire component.

> I assume it's down to the order of the classifiers in the array.

That's right. The higher up the classifier in the array, the higher the priority.

**Comment by ryanshipswell** [comment]
Added PR #201 for this.
